services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/ApiBlockingFilter.kt

```diff
@@ -0,0 +1,51 @@
+package fi.hel.haitaton.hanke
+
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import mu.KotlinLogging
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * In order to do maintenance tasks, we need to be able to block calls to the API and return 503
+ * Service Unavailable.
+ */
+@Order(1)
+@Component
+@ConditionalOnProperty("haitaton.api.disabled")
+class ApiBlockingFilter : OncePerRequestFilter() {
+
+    override fun initFilterBean() {
+        super.initFilterBean()
+        log.info { "API blocking filter initialized" }
+    }
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        log.info { "API is offline, request blocked: ${request.method} ${request.requestURI}" }
+        response.status = HttpServletResponse.SC_SERVICE_UNAVAILABLE
+    }
+
+    override fun shouldNotFilter(request: HttpServletRequest): Boolean =
+        ALLOWED_PATHS.any { request.requestURI.matches(it.toRegex()) }
+
+    companion object {
+        val ALLOWED_PATHS =
+            setOf(
+                "/actuator/.*",
+                "/status",
+                "/swagger-ui.html",
+                "/swagger-ui/.*",
+                "/v3/api-docs.*",
+                "/profiili/.*",
```

Comment:
Why are these allowed? Especially `/profiili`.

Reply:
What should we allow?

Reply:
We need to allow `/status`, `/actuator/health/liveness` and `/actuator/health/readiness` so the docker pods are considered active and we generate extra errors in monitoring.

I don't know of any reasons to allow anything else.

Reply:
Right you are. I changed the white-listing to allow only these endpoints.